rtl/eclrtl/rtlfield.cpp

```diff
@@ -1609,7 +1643,13 @@ size32_t RtlUtf8TypeInfo::toXML(const byte * self, const byte * selfrow, const R
     unsigned thisLength = rtlReadUInt4(self);
     unsigned thisSize = sizeof(size32_t) + rtlUtf8Size(thisLength, str);
 
-    target.outputUtf8(thisLength, str, queryScalarXPath(field));
+    if (field->hasInlineContentXpath())
+    {
+        target.outputInline(thisLength, str, field->hasNamedContentXpath() ? queryScalarXPath(field) : nullptr);
```

Comment:
Need to be careful about length v size.  (Worth naming size variables appropriately.)